sbt-scala-native/src/main/scala/scala/scalanative/sbtplugin/ScalaNativePluginInternal.scala

```diff
@@ -131,7 +132,9 @@ object ScalaNativePluginInternal {
           .withCompilerConfig(nativeConfig.value)
       }
 
-      interceptBuildException(Build.build(config, outpath.toPath))
+      sharedScope.identifiedBy(config) { implicit scope =>
```

Comment:
Different projects can different `config`s and yet access the same jars. Identifying the scope by config is not going to solve the issue in all cases.

Why did you want an identifying key? Can't we simply use a single slot? (like we have a single slot for the allocated test adapters)

Reply:
> Why did you want an identifying key? Can't we simply use a single slot? (like we have a single slot for the allocated test adapters)

I thought that when uses `sbt nativeLink` and has multiple native subprojects it would call  `onComplete` for each of them, but apparently it would be called only once. It was only used in order to mark usages coming for different threads. I'll fix that as it is redundant.